src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -0,0 +1,87 @@
+            + "to the event specified by the index number used in the displayed event list.\n"
+            + "Parameters: RESIDENT_INDEX EVENT_INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1 1";
+
+    public static final String MESSAGE_INVALID_EVENT = "The event provided is invalid";
+    public static final String MESSAGE_ASSIGN_PERSON_SUCCESS = "Assigned resident %s to %s";
+    public static final String MESSAGE_DUPLICATE_PERSON_ADDED = "Duplicate resident %s being added to event %s";
+
+    // index of person in person list to add
+    // event that person will be added to
+    // note that event is just a copy and not the direct reference to the event in the event list in model
+    private final Index residentIndex;
+    private final Index eventIndex;
+
+    /**
+     * @param residentIndex index of the resident to be added
+     * @param eventIndex index of the event to be modified
+     */
+    public AssignCommand(Index residentIndex, Index eventIndex) {
+        this.residentIndex = residentIndex;
+        this.eventIndex = eventIndex;
+    }
+
+    // an example of an assign command is assign 1 Hall dinner
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Event> eventList = model.getEventList();
+
+        // get person from list based on index
+        if (residentIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        if (eventIndex.getZeroBased() >= eventList.size()) {
+            throw new CommandException(MESSAGE_INVALID_EVENT);
+        }
+
+        // ensure that index is not negative
+        // to be taken out on final release
+        assert residentIndex.getZeroBased() >= 0;
+        assert eventIndex.getZeroBased() >= 0;
+
+        Person personToAdd = lastShownList.get(residentIndex.getZeroBased());
+
+        // get event from list
+        Event event = eventList.get(eventIndex.getZeroBased());
+
+        // get attendee list from event
+        Set<Person> attendeesList = new HashSet<>(event.getAttendeesList());
+
+        // check if person to add is already in the event list
+        if (attendeesList.contains(personToAdd)) {
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_PERSON_ADDED,
+                    personToAdd.getName(),
+                    event.getName()));
+        }
+
+        // add person to event's attendee list
+        attendeesList.add(personToAdd);
+        Event editedEvent = new Event(event.getName(), event.getDescription(), attendeesList);
+
+        // TODO: update model's observable list of events similar to ModelManager#updateFilteredPersonList()
+        // when updating observable list of persons
+        model.setEvent(event, editedEvent);
```

Comment:
Hmm could you explain the TODO? 

Based on my understanding, `ModelManager` updates the `filteredPersonList` during a `add` command because the user may have keyed in `find X`, followed by `add Y` - so the UI needs to reset back to the full view by updating the predicate.

Reply:
This is an outdated TODO. I have removed it in the latest commit.